src/operations.jl

```diff
@@ -1,7 +1,18 @@
+"""
+    ArrayReference
+
+A type for encoding an array reference `A[i,j]` occurring inside an `@avx` block.
+
+# Fields
 
+$(TYPEDFIELDS)
+"""
 struct ArrayReference
+    "The array variable"
     array::Symbol
+    "The list of indices (e.g., `[:i, :j]`). Not sure what this does for computed indices, maybe that chains with Operations."
     indices::Vector{Symbol}
```

Comment:
This should probably be combined with `ArrayReferenceMeta`, because it doesn't provide enough information by itself.
 
For computed indices, it will either
1. If the index is computed as a constant offset representable by an `Int8`, it'll set the corresponding index in `offsets` to that offset.
2. Define an `Operation` performing the computation. `name(op)` will be the corresponding index of `indices`.